qutip/bloch.py

```diff
@@ -401,6 +411,177 @@ def add_annotation(self, state_or_vector, text, **kwargs):
 
+    def add_latitude(self, init_pt, fin_pt):
+        # connects two points with an arc on a same latitude.
+        # Inputs have to be in spherical coordinates.
+        # Used in the function add_arc        
+        if abs(init_pt[2] - fin_pt[2]) > pi:
+            if init_pt[2] <= fin_pt[2]:
+                init_pt[2] = 2*pi + init_pt[2]
+            else:
+                fin_pt[2] = 2*pi + fin_pt[2]
+          
+        phi = linspace(init_pt[2], fin_pt[2], 200)
+
+        r = init_pt[0]
+        theta = linspace(init_pt[1], init_pt[1], 200)
+
+        arc_x = r*sin(theta)*cos(phi)
+        arc_y = r*sin(theta)*sin(phi)
+        arc_z = r*cos(theta)
+
+        self.arcs_x.append(arc_x)
+        self.arcs_y.append(arc_y)
+        self.arcs_z.append(arc_z)
+
+    def add_longitude(self, init_pt, fin_pt):
+        # connects two points with an arc on a same longitude.
+        # Inputs have to be in spherical coordinates.
+        # Used in the function add_arc
+        r = init_pt[0]
+        phi = linspace(fin_pt[2], fin_pt[2], 200)
+        theta = linspace(init_pt[1], fin_pt[1], 200)
+
+        arc_x = r*sin(theta)*cos(phi)
+        arc_y = r*sin(theta)*sin(phi)
+        arc_z = r*cos(theta)
+
+        self.arcs_x.append(arc_x)
+        self.arcs_y.append(arc_y)
+        self.arcs_z.append(arc_z)
+
+    def cart_to_sph(self, cart_cord):
+        # converts cartesian coordinates to spherical
+        r = 0
+        for i in cart_cord:
+            r += i**2
+        r = sqrt(r)
+
+        if r == 0:
+            print("Polar and azimuthal points not defined for origin")
+            return
+
+        theta = arccos(cart_cord[2]/r)
+
+        if theta < 1e-12 or abs(theta - pi) < 1e-12:
+            phi = 0
+        else:
+            x = cart_cord[0]
+            y = cart_cord[1]
+            if x == 0:
+                if y > 0:
+                    phi = pi/2
+                else:
+                    phi = pi + pi/2
+            elif x > 0 and y >= 0:
+                phi = arctan(y/x)
+            elif x < 0 and y >= 0:
+                phi = pi - arctan(abs(y)/abs(x))
+            elif x < 0 and y < 0:
+                phi = pi + arctan(abs(y)/abs(x))
+            else:
+                phi = 2*pi - arctan(abs(y)/abs(x))
+        return [r, theta, phi]
+
+    def rot_matz(self, angle):
+        # Rotation matrix about z-axis
+        rot_matz_ = R.from_matrix([[cos(angle), -sin(angle), 0],
+                                   [sin(angle),  cos(angle), 0],
+                                   [         0,           0, 1]])
+        return rot_matz_
+
+    def rot_maty(self, angle):
+        # Rotation matrix about y-axis
+        rot_maty_ = R.from_matrix([[cos(angle), 0, -sin(angle)],
+                                   [         0, 1,           0],
+                                   [sin(angle), 0,  cos(angle)]])
+        return rot_maty_
+
+    def add_arc(self, init_pt, fin_pt):
```

Comment:
Right, but if we are letting the user give the input, I thought it would be easy and better for the user to specify the cartesian coordinates rather than spherical ones. What do you think?

Reply:
A short notice. We should be consistent with other methods of `Bloch`. It seems that e.g. `add_points` accepts input in the cartesian coordinate?

Reply:
> A short notice. We should be consistent with other methods of `Bloch`. It seems that e.g. `add_points` accepts input in the cartesian coordinate?

Thanks, Boxi. Yes, `add_points` accepts input in Cartesian coordinates. So, it is fine to let the input of `add_arc` be in cartesian coordinates too, right?

Reply:
Lets leave the cartesian coordinates then.